File: unit-2/sprint-1/day-6-Event Listener & Saving UI State/assignments/app/src/main/java/com/example/intentfilter/MainActivity.java
package com.example.intentfilter;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
     TextView mTvView;
     EditText mEtEnterMessage;
     Button mBtnClick;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mTvView = findViewById(R.id.tvView);
        mBtnClick = findViewById(R.id.btnClick);
        mEtEnterMessage = findViewById(R.id.etEnterMessage);
        mBtnClick.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String str = mEtEnterMessage.getText().toString();
                mTvView.setText(str);
            }
        });
        mEtEnterMessage.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                Log.d("Tag", "before text" + s.toString());
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                Log.d("Tag", "on text" + s.toString());
            }

            @Override
            public void afterTextChanged(Editable s) {
                Log.d("Tag", "after text" + s.toString());
                if (s.toString().length() >= 6) {
                    Toast.makeText(MainActivity.this, s.toString(), Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Coding Preparation Series/Search The Item/app/src/main/java/com/example/search/MainActivity.kt
package com.example.search

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Toast
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.search.api.CityAdaptor
import com.example.search.data.Address
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var cityViewModel: CityViewModel
    private var list = emptyList<Address>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
       setContentView(R.layout.activity_main)
        cityViewModel = ViewModelProviders.of(this).get(CityViewModel::class.java)
        search()
    }

    private fun search() {
        et_Search.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
                loadapi(p0.toString())
            }

            override fun afterTextChanged(p0: Editable?) {
            }

        })

    }

    private fun loadapi(query: String) {
        cityViewModel.liveData.observe(this, {
            it.let {
                when (it) {
                    is MainUiModel.OnSuccess -> {
                            list = it.responseDTO.data.addressList as ArrayList<Address>
                            val adaptor = CityAdaptor(list)
                            val layoutManager = LinearLayoutManager(this)
                            recyclerView.adapter = adaptor
                            recyclerView.layoutManager = layoutManager
                        }

                    is MainUiModel.OnFailure ->{
                        Toast.makeText(this, "Error", Toast.LENGTH_SHORT).show()
                    }
                }
            }
        })
        cityViewModel.getData(query)
    }
}